feat: removing the application stage 2

deborahgu opened the pull request:
This removes the deprecated demographics django application. No other apps in the repository currently reference this djangoapp in code or tables.

This DEPR was accepted in https://github.com/openedx/edx-platform/issues/35127. Only 2U ever used this app.

* removing the application folder
* removing references in all other files

FIXES: APER-3560

deborahgu committed:
feat: removing the application stage 2

* removing the application folder
* removing references in all other files

FIXES: APER-3560

deborahgu committed:
feat:  removing demographics from test shards

removing an entire app.

FIXES: APER-3560

deborahgu committed:
feat:  updating the github action

actions need to know about changes to djangoapps

FIXES: APER-3560

justinhynes commented on scripts/user_retirement/utils/edx_api.py:
```diff
@@ -474,28 +474,6 @@ def replace_usernames(self, username_mappings):
         return self._request('POST', api_url, json=data)
 
 
-class DemographicsApi(BaseApiClient):
```
This particular endpoint wasn't part of the Demographics app in edx-platform... wouldn't we still need to support retiring learners in the Demographics IDA itself?

I don't know enough about the inner workings of the retirement pipeline yet to know if it's using this particular piece of code. It appears that the DemographicsAPI is still part of the (forked) `tubular` repo in our GitHub org... so maybe we're okay removing this one? 

Do you have a better understanding of the spaghetti?

deborahgu replied:
Aren't we about to retire the whole IDA, though? Does it matter if we miss a couple of retirement requests in the window before we delete all the data?

deborahgu replied:
on second thought, Justin is right, the DEPR of the IDA is soon but it's not _imminent_, better safe than sorry.

deborahgu committed:
feat:  this references the IDA, which has not been deprecated yet

undoing a removal from the retirement pipeline that we aren't quite ready for yet

FIXES: APER-3560

deborahgu committed:
Merge branch 'master' into dkaplan1/APER-3560_depr-remove-demographics-django-app-from-edx-platform-pt2

justinhynes reviewed (approved):
Thanks for talking through the changes! LGTM.
